psi_cardinality/c/psi_cardinality_benchmark_c.cpp

```diff
@@ -0,0 +1,191 @@
+#include "absl/strings/str_cat.h"
+#include "benchmark/benchmark.h"
+#include "psi_cardinality_client_c.h"
+#include "psi_cardinality_server_c.h"
+
+namespace psi_cardinality {
+namespace {
+
+void BM_ServerBindingSetup(benchmark::State &state, double fpr) {
+  psi_cardinality_server_ctx server_;
+  char *err;
+  psi_cardinality_server_create_with_new_key(&server_, &err);
+
+  int num_inputs = state.range(0);
+  int num_client_inputs = 10000;
+  std::vector<std::string> orig_inputs(num_inputs);
+  std::vector<server_buffer_t> inputs(num_inputs);
+
+  for (int i = 0; i < num_inputs; i++) {
+    orig_inputs[i] = absl::StrCat("Element", i);
+    inputs[i] = {orig_inputs[i].c_str(), orig_inputs[i].size()};
+  }
+  std::string setup;
+  int64_t elements_processed = 0;
+  for (auto _ : state) {
+    char *server_setup = nullptr;
+    size_t server_setup_buff_len = 0;
+    psi_cardinality_server_create_setup_message(
+        server_, fpr, num_client_inputs, inputs.data(), inputs.size(),
+        &server_setup, &server_setup_buff_len);
+
+    ::benchmark::DoNotOptimize(
+        std::string(server_setup, server_setup_buff_len));
```

Comment:
This creates an unnecessary copy of the string. The following should do the trick without copying.
```C++
::benchmark::DoNotOptimize(server_setup);
::benchmark::ClobberMemory();
```
See also: https://github.com/google/benchmark/#preventing-optimization